src/System.Linq/src/System/Linq/Skip.cs

```diff
@@ -121,5 +121,33 @@ private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSour
                 }
             }
         }
+
+#if netcoreapp11
+        public static IEnumerable<TSource> SkipLast<TSource>(this IEnumerable<TSource> source, int count)
+        {
+            if (source == null)
+            {
+                throw Error.ArgumentNull(nameof(source));
+            }
+
+            return SkipLastIterator(source, Math.Max(count, 0));
+        }
+
+        private static IEnumerable<TSource> SkipLastIterator<TSource>(IEnumerable<TSource> source, int count)
+        {
+            var queue = new Queue<TSource>();
```

Comment:
@ViIvanov Thanks for noticing that; it was a problem with `TakeLast` as well. Fixed for both methods.